src/main/java/com/aws/greengrass/clientdevices/auth/configuration/CAConfiguration.java

```diff
@@ -104,17 +117,25 @@ private static CertificateStore.CAType getCaTypeFromConfiguration(Topics configu
         return CertificateStore.CAType.valueOf(caType);
     }
 
-    private static String getCaPrivateKeyUriFromConfiguration(Topics certAuthorityTopic) {
-        // NOTE: Parse the key to ensure it is a valid URI
-        //  before returning it
-        Topic privateKeyUriTopic = certAuthorityTopic.find(CA_PRIVATE_KEY_URI);
-        return Coerce.toString(privateKeyUriTopic);
+    private static Optional<URI> getCaPrivateKeyUriFromConfiguration(Topics certAuthorityTopic) throws
+            URISyntaxException {
+        String maybePrivateKeyUri = Coerce.toString(certAuthorityTopic.findOrDefault("", CA_PRIVATE_KEY_URI));
+
+        if (Utils.isEmpty(maybePrivateKeyUri)) {
+            return Optional.empty();
+        }
+
+        return Optional.of(new URI(maybePrivateKeyUri));
     }
 
-    private static String getCaCertificateUriFromConfiguration(Topics certAuthorityTopic) {
-        // NOTE: Parse the key to ensure it is a valid URI
-        //  before returning it
-        Topic privateKeyUriTopic = certAuthorityTopic.find(CA_CERTIFICATE_URI);
-        return Coerce.toString(privateKeyUriTopic);
+    private static Optional<URI> getCaCertificateUriFromConfiguration(Topics certAuthorityTopic) throws
```

Comment:
Changed this method to parse the configuration string into a valid URI, it returns optional since it could be well... optional...